From 38e3c6ebbcd9e1d0481889336cb62a7d237f389f Mon Sep 17 00:00:00 2001
From: Eric Caruso <ejcaruso@chromium.org>
Date: Tue, 18 Nov 2014 11:24:01 -0800
Subject: [PATCH] CHROMIUM: i915: prevent brightness from jumping up on enable

When a user dims the screen all the way, waits for a few seconds
so the backlight is disabled, and then raises the brightness, the
screen would flash on for a moment before the userspace daemons
changed it to the level it should be.

The i915 driver seems to be enforcing some brightness policy that
we don't like, though it would almost certainly break someone's
userspace if we floated this upstream.

Note:
[drm:intel_panel_actually_set_backlight] set backlight PWM = 0
[drm:intel_edp_backlight_power] panel power control backlight disable
[drm:intel_backlight_device_update_status] updating intel_backlight, brightness=0/19200
[drm:intel_panel_actually_set_backlight] set backlight PWM = 0
[drm:intel_edp_backlight_off]
[drm:intel_panel_actually_set_backlight] set backlight PWM = 0
[drm:intel_edp_backlight_on]
[drm:intel_panel_enable_backlight] pipe A
[drm:intel_panel_actually_set_backlight] set backlight PWM = 19200

BUG=b:115403276
TEST=Repeatedly lower down brightness by press/release F6 key,
examine display brightness remains low without temporary full
brightness. i.e. monitoring kernel logs w/ drm.debug=0xe, no
policy override as maximum like notes above, in sequence of:
PWM is set to 0, backlight off then on.

Signed-off-by: Eric Caruso <ejcaruso@chromium.org>
(cherry picked from commit 269a9f390c04ba4e8e0defa6421b0f7a4e973dbf
of the chromeos-v4.4)
Signed-off-by: Harry Pan <harry.pan@intel.com>
Change-Id: I54366b97e40368d0731fbd1d3bb0eefd181bbc17
Reviewed-on: https://chromium-review.googlesource.com/1226732
Reviewed-by: Aaron Durbin <adurbin@chromium.org>
Reviewed-by: Justin TerAvest <teravest@chromium.org>
---
 drivers/gpu/drm/i915/intel_panel.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 4a9f139e7b73..72d766a7fc7e 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -1105,7 +1105,6 @@ void intel_panel_enable_backlight(const struct intel_crtc_state *crtc_state,
 	WARN_ON(panel->backlight.max == 0);
 
 	if (panel->backlight.level <= panel->backlight.min) {
-		panel->backlight.level = panel->backlight.max;
 		if (panel->backlight.device)
 			panel->backlight.device->props.brightness =
 				scale_hw_to_user(connector,
-- 
2.23.0.rc1.153.gdeed80330f-goog

